src/openklant/components/klantinteracties/models/validators.py

```diff
@@ -0,0 +1,156 @@
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+from vng_api_common.validators import (
+    BaseIdentifierValidator,
+    validate_bsn,
+    validate_rsin,
+)
+
+from .constants import (
+    PartijIdentificatorCodeObjectType,
+    PartijIdentificatorCodeRegister,
+    PartijIdentificatorCodeSoortObjectId,
+)
+
+VESTIGINGSNUMMER_LENGTH = 12
+KVK_NUMMER_LENGTH = 8
+
+
+class PartijIdentificatorValidator:
+
+    NATUURLIJK_PERSOON = [
+        PartijIdentificatorCodeSoortObjectId.bsn.value,
+        PartijIdentificatorCodeSoortObjectId.overig.value,
+    ]
+    VESTIGING = [
+        PartijIdentificatorCodeSoortObjectId.vestigingsnummer.value,
+        PartijIdentificatorCodeSoortObjectId.overig.value,
+    ]
+    NIET_NATUURLIJK_PERSOON = [
+        PartijIdentificatorCodeSoortObjectId.rsin.value,
+        PartijIdentificatorCodeSoortObjectId.kvk_nummer.value,
+        PartijIdentificatorCodeSoortObjectId.overig.value,
+    ]
+
+    ALLOWED_OBJECT_TYPES_FOR_CODE_OBJECTTYPE = {
+        PartijIdentificatorCodeObjectType.natuurlijk_persoon.value: NATUURLIJK_PERSOON,
+        PartijIdentificatorCodeObjectType.vestiging.value: VESTIGING,
+        PartijIdentificatorCodeObjectType.niet_natuurlijk_persoon.value: NIET_NATUURLIJK_PERSOON,
+    }
+
+    ALLOWED_OBJECT_TYPES_FOR_REGISTRIES = {
+        PartijIdentificatorCodeRegister.brp: {
+            PartijIdentificatorCodeObjectType.natuurlijk_persoon.value: NATUURLIJK_PERSOON,
+            PartijIdentificatorCodeObjectType.overig.value: [],
```

Comment:
I'm not sure if `overig` needs to be here, since the methods below return early if register == `overig`